lib/azure/armrest/network/ip_address_service.rb

```diff
@@ -0,0 +1,102 @@
+module Azure
+  module Armrest
+    # Class for managing public IP addresss.
+    class IpAddressService < ArmrestService
+
+      # Creates and returns a new IpAddressService instance.
+      #
+      def initialize(_armrest_configuration, options = {})
+        super
+        @provider = options[:provider] || 'Microsoft.Network'
+        set_service_api_version(options, 'publicIPAddresses')
+      end
+
+      # Return information for the given IP address name for the
+      # provided +resource_group+. If no group is specified, it will use
+      # the resource group set in the constructor.
+      #
+      # Example:
+      #
+      #   # Where 'your_ip_name' likely corresponds to a VM name.
+      #   ip.get('your_ip_name', 'your_resource_group')
+      #
+      def get(ip_name, resource_group = armrest_configuration.resource_group)
+        raise ArgumentError, "must specify resource group" unless resource_group
+        url = build_url(resource_group, ip_name)
+        JSON.parse(rest_get(url))
+      end
+
+      # Shortcut method that returns just the IP address for the given public
+      # IP address name.
+      #
+      def get_ip(ip_name, resource_group = armrest_configuration.resource_group)
+        get(ip_name, resource_group)['properties']['ipAddress']
+      end
+
+      # Returns a list of available IP addresss for the given subscription
+      # for the provided +group+, or for all resource groups if no group is specified.
+      #
+      def list(group = nil)
+        if group
+          url = build_url(group)
+          JSON.parse(rest_get(url))['value']
+        else
+          array = []
+          threads = []
+          mutex = Mutex.new
+
+          resource_groups.each do |rg|
+            threads << Thread.new(rg['name']) do |group|
+              url = build_url(group)
+              response = rest_get(url)
+              results = JSON.parse(response)['value']
+              if results && !results.empty?
+                mutex.synchronize{
+                  results.each{ |hash| hash['resourceGroup'] = group }
+                  array << results
+                }
+              end
+            end
+          end
+
+          threads.each(&:join)
+
+          array.flatten
+        end
+      end
+
+      # List all IP addresss for the current subscription.
+      #
+      def list_all_for_subscription
```

Comment:
As far as I can tell, yes. They seem to have added support for per-subscription calls for just about everything now, though it wasn't always so. However, it doesn't explicitly include resource group information, which could be an issue. We'd have to parse it out of an id field. That's something we could add to the base json model class once we modify the code to return them instead of plain hashes.